domains/Discussions/Tests/Feature/QuestionsIndexTest.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+namespace Domains\Discussions\Tests\Feature;
+
+use Carbon\Carbon;
+use Domains\Accounts\Database\Factories\UserFactory;
+use Domains\Accounts\Models\User;
+use Domains\Discussions\Database\Factories\QuestionFactory;
+use Illuminate\Http\Response;
+use Laravel\Lumen\Testing\DatabaseMigrations;
+use Tests\TestCase;
+
+class QuestionsIndexTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    private User $user;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->user = UserFactory::new()->create();
+        QuestionFactory::times(10)->create();
+    }
+
+    /** @test */
+    public function it_list_non_deleted_question(): void
+    {
+        $deleteQuestion = QuestionFactory::new([
+            'deleted_at' => Carbon::now(),
+        ])
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index'))
+            ->seeJsonStructure([
+                'data' => [
+                    [
+                        'id',
+                        'title',
+                        'slug',
+                        'description',
+                        'author',
+                        'created_at',
+                        'updated_at',
+                        'resolved_at',
+                        'deleted_at',
+                    ]
+                ],
+                'links' => [
+                    'first', 'prev', 'next', 'last',
+                ],
+            ])
+            ->seeJsonDoesntContains([
+                'email' => $deleteQuestion->author->email,
+            ])
+            ->seeJsonContains([
+                'to' => 10,
+            ]);
+    }
+
+    /** @test */
+    public function it_blocks_guest_for_many_attempts(): void
+    {
+        for ($attempt = 0; $attempt < 30; ++$attempt) {
+            $this->get(route('discussions.questions.index'))
+                ->assertResponseStatus(Response::HTTP_OK);
```

Comment:
There's no need to check for the response status code in this loop.

Reply:
Hi @josepostiga I've discussed this with @dleicam and I was the one asking for it since we want to make sure `Too many requests` is sent only when exceeding 30 req/min. If we don't assert an `OK` response then the test will still pass even if limit to 1 req/min.